llvm/include/llvm/IR/InstrTypes.h

```diff
@@ -946,6 +946,56 @@ class PossiblyNonNegInst : public CastInst {
   }
 };
 
+/// Cast Instruction that can have a nowrap flags (only trunc)
+class PossiblyNoWrapInst : public CastInst {
+public:
+  enum { AnyWrap = 0, NoUnsignedWrap = (1 << 0), NoSignedWrap = (1 << 1) };
+
+  static bool classof(const Instruction *I) {
+    return I->getOpcode() == Instruction::Trunc;
+  }
+
+  static bool classof(const Value *V) {
+    return isa<Instruction>(V) && classof(cast<Instruction>(V));
```

Comment:
dyn_cast instead of isa+cast?

Reply:
In the internal `classof` there's a check `I->getOpcode() == Instrucntion::Trunc`, 
won't `dyn_cast` cause a null pointer dereference?